datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -198,6 +198,7 @@ impl<S: SimplifyInfo> ExprSimplifier<S> {
     ///
     /// See [Self::simplify] for details and usage examples.
     ///
+    #[deprecated(since = "48.0.0", note = "")]
```

Comment:
You could reduce the amount of code duplication here by rewriting this function to be in terms of `simplify_with_cycle_count`. Something like:

```rust
    #[deprecated(since = "48.0.0", note = "")]
    pub fn simplify_with_cycle_count(&self, expr: Expr) -> Result<(Expr, u32)> {
        let (transformed, cycle_count) =
            self.simplify_with_cycle_count(expr)?;
        Ok((transformed.data, cycle_count))
    }
```

Reply:
Great point!